client/src/app/user-detail/user-detail.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+
+import { UserService } from '../user.service';
+
+import{ User } from '../fluff/user';
+
+
+@Component({
+  selector: 'app-user-detail',
+  templateUrl: './user-detail.component.html',
+  styleUrls: ['./user-detail.component.scss']
+})
+export class UserDetailComponent implements OnInit {
+
+  // @Input() user?: {key: number, value: User};
+  public user?: {key: number, value: User};
+  // user?: [number, User];
+
+  constructor(
+    private route: ActivatedRoute,
+    private userService: UserService,
+    private location: Location
+  ) { }
+
+  public ngOnInit(): void {
+    this.getUser();
+  }
+
+
+  private getUser(): void {
+    const id = Number(this.route.snapshot.paramMap.get('id'));
+    this.userService.getUser(id)
+      .subscribe(user => this.user = user);
```

Comment:
In general: If you subscribe to something within a component, then ensure that the subscription is cleared when the component is going to be destroyed